wp-cli/vip-fixers.php

```diff
@@ -1,48 +1,110 @@
 <?php
 
 class VIP_Go_OneTimeFixers_Command extends WPCOM_VIP_CLI_Command {
+	public function enforce_govip_ssl( $args, $assoc_args ) {
+		$results = [];
 
-	private function wpcom_vip_download_image( $image_url, $post_id = 0, $description = '' ) {
-		if ( $post_id < 0 ) {
-			return new WP_Error( 'invalid-post-id', 'Please specify a valid post ID.' );
-		}
+		if ( is_multisite() ) {
+			$site_ids = get_sites( [
+				'number' => 99999,
+				'fields' => 'ids',
+			] );
+
+			foreach ( $site_ids as $site_id ) {
+				switch_to_blog( $site_id );
+
+				$old_home_url = home_url();
+
+				$updated = $this->enforce_govip_ssl_for_current_site();
+
+				$results[ $old_home_url ] = [
+					// Get from database to avoid set_url_scheme and filters
+					'home' => $this->get_raw_option_value( 'home' ),
+					'siteurl' => $this->get_raw_option_value( 'siteurl' ),
+					'updated' => $updated,
+				];
 
-		if ( ! filter_var( $image_url, FILTER_VALIDATE_URL ) ) {
-			return new WP_Error( 'not-a-url', 'Please specify a valid URL.' );
+				refresh_blog_details();
+				restore_current_blog();
+			}
+
+			$this->flush_cache();
+		} else {
+			$old_home_url = home_url();
+
+			$updated = $this->enforce_govip_ssl_for_current_site();
+
+			$results[ $old_home_url ] = [
+				// Get from database to avoid set_url_scheme and filters
+				'home' => $this->get_raw_option_value( 'home' ),
+				'siteurl' => $this->get_raw_option_value( 'siteurl' ),
+				'updated' => $updated,
+			];
 		}
 
-		$image_url_path = parse_url( $image_url, PHP_URL_PATH );
-		$image_path_info = pathinfo( $image_url_path );
+		foreach ( $results as $old_home_url => $result ) {
+			$home = $result['home'];
+			$siteurl = $result['siteurl'];
+			list( $updated_home, $updated_siteurl ) = $result['updated'];
+
+			$message = sprintf(
+				'Finished for %s; home updated? %s (%s) | siteurl updated? %s (%s)',
+				$old_home_url,
+				var_export( $updated_home, true ),
+				$home,
+				var_export( $updated_siteurl, true ),
+				$siteurl
+			);
+
+			WP_CLI::log( '#vip-go-https-cleanup: ' . $message );
 
-		if ( ! in_array( strtolower( $image_path_info['extension'] ), array( 'avi', 'mp3', 'jpg', 'jpe', 'jpeg', 'gif', 'png' ) ) ) {
-			return new WP_Error( 'not-an-image', 'Specified URL does not have a valid media extension.' );
+			wpcom_vip_irc( '#vip-go-https-cleanup', $message );
 		}
+	}
 
-		// Download file to temp location; short timeout, because we don't have all day.
-		$downloaded_url = download_url( $image_url, 30 );
+	private function enforce_govip_ssl_for_current_site() {
+		$updated_home = $this->enforce_govip_ssl_for_option_url( 'home' );
+		$updated_siteurl = $this->enforce_govip_ssl_for_option_url( 'siteurl' );
 
-		// We couldn't download and store to a temporary location, so bail.
-		if ( is_wp_error( $downloaded_url ) ) {
-			return $downloaded_url;
+		if ( $updated_home || $updated_siteurl ) {
+			$this->flush_cache( get_site_url() );
```

Comment:
Since this happens in the site context, couldn't we just call `wp_cache_flush()`? Does running it through a second WP-CLI process solve some problem?